aten/src/ATen/native/Normalization.cpp

```diff
@@ -461,7 +461,9 @@ std::tuple<Tensor, Tensor, Tensor> _batch_norm_impl_index_backward(
   const Tensor& save_mean = c10::value_or_else(save_mean_opt, [] {return Tensor();});
   const Tensor& save_var_transform = c10::value_or_else(save_var_transform_opt, [] {return Tensor();});
 
-  if (impl_index == 0) {
+  // backward in inference mode is not supported in cudnn, fallback to native
```

Comment:
Was this failing before this PR?
If so, could we add a test in test_nn.py that makes sure that we don't introduce this again in the future.

Reply:
I vaguely remember that we run into a failing case in the test added below. Nick has mentioned about moving that test into meta test set: https://github.com/pytorch/pytorch/pull/57321/files/2855c79bbd43fc2135fc238b96d3a19173b5567b#r670053319

Reply:
Ok, we should make sure we have a test for it in this PR.

Reply:
I verified the failure and fix in the updated python test.

Reply:
I don't see any update to `test_nn` here corresponding to that?

Reply:
this is covered in https://github.com/pytorch/pytorch/pull/57321/files#diff-c54c3bae2ea206c4de5b90adf7f2d3ec47df6818451b8dd2b3c36fe54adc10d8R10759

That's probably not a good place for the purpose of the test. I'll move it to `test_nn.py` with a simplified version, since the majority of the old test is covered in generated jit metaprogramming test.

Reply:
Sorry, I was wrong earlier. I don't think this fix applies to test_nn.

Autograd has already handled this, when we run with inference, we fallback to native_batch_norm_backward. https://github.com/pytorch/pytorch/blob/master/tools/autograd/derivatives.yaml#L2118

This is exactly what we are doing here for autodiff. And I do think `test_jit.py` is the right place to verify that fix.